Library/Homebrew/formulary.rb

```diff
@@ -14,10 +14,14 @@ def self.formula_class_get(path)
     FORMULAE.fetch(path)
   end
 
-  def self.load_formula(name, path)
+  def self.load_formula(name, path, io=nil)
     mod = Module.new
     const_set("FormulaNamespace#{Digest::MD5.hexdigest(path.to_s)}", mod)
-    contents = path.open("r") { |f| set_encoding(f).read }
+    if io
+      contents = set_encoding(io).read
+    else
+      contents = path.open("r") { |f| set_encoding(f).read }
+    end
```

Comment:
Since we always know the type we are passing in, I would just make a separate method rather than branching here. For example, `load_formula` could just accept an IO, and then

```rb
def self.load_formula_from_path(name, path)
  path.open("r") { |f| load_formula(name, f) }
end
```